specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2023-12-10-preview/cloudhsm.json

```diff
@@ -0,0 +1,967 @@
+            "$ref": "#/parameters/peConnectionName"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Private endpoint connection successfully returned.",
+            "schema": {
+              "$ref": "#/definitions/PrivateEndpointConnection"
+            }
+          },
+          "default": {
+            "description": "The error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudHsmClusterError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "CloudHsmClusterPrivateEndpointConnection_Get_MaximumSet_Gen": {
+            "$ref": "./examples/CloudHsmClusterPrivateEndpointConnection_Get_MaximumSet_Gen.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HardwareSecurityModules/cloudHsmClusters/{cloudHsmClusterName}/privateEndpointConnections": {
+      "get": {
+        "tags": [
+          ""
+        ],
+        "operationId": "PrivateEndpointConnections_ListByCloudHsmCluster",
+        "description": "The List operation gets information about the private endpoint connections associated with the Cloud HSM Cluster",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/cloudHsmClusterName"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "List of all CloudHsmCluster private endpoint connections associated with the Cloud HSM Cluster.",
+            "schema": {
+              "$ref": "#/definitions/PrivateEndpointConnectionListResult"
+            }
+          },
+          "default": {
+            "description": "The error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudHsmClusterError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "CloudHsmClusterPrivateEndpointConnection_ListByResource_MaximumSet_Gen": {
+            "$ref": "./examples/CloudHsmClusterPrivateEndpointConnection_ListByCloudHsmCluster_MaximumSet_Gen.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "CloudHsmProperties": {
+      "type": "object",
+      "description": "The Cloud HSM Properties",
+      "properties": {
+        "fqdn": {
+          "type": "string",
+          "description": "FQDN of the Cloud HSM"
+        },
+        "state": {
```

Comment:
Is there anywhere I can review the possible states ?

Reply:
@sumantshiv, I believe we get this state from the HFC in the HSMPool spec and it is a read only field that we return for the customer as part of the response. These values are not controlled by us and we cannot add the list of the expected values because any time HFC makes a change for this property it will affect our Swagger documentation.